src/routes/sessions/[id]/+page.svelte

```diff
@@ -54,6 +55,23 @@
 	$: if ($settingsStore?.ollamaModel) session.model = $settingsStore.ollamaModel;
 	$: knowledge = knowledgeId ? loadKnowledge(knowledgeId) : null;
 	$: shouldFocusTextarea = !isPromptFullscreen;
+	$: {
+		if (session && session.id !== $currentSessionId) {
+			getModelsList();
+			currentSessionId.set(session.id);
+		}
+	}
+
+	async function getModelsList() {
+		if (!$settingsStore) return;
+		try {
+			$settingsStore.ollamaModels = (await ollamaTags()).models;
+			//TODO hide error message if present
+		} catch {
+			$settingsStore.ollamaModels = [];
+			handleError(new Error('Failed to fetch'));
+		}
+	}
```

Comment:
I don't think we want to call `handleError()` here, it causes a bug when creating a **New session** and Ollama is offline it renders the error in the as a session message, that triggers the the model drop-down and knowledge fields to disappear (as if it's no longer a "new session").

Instead I'm thinking we could do something like this:
```svelte
<svelte:fragment slot="label">
	Prompt

	{#if !$settingsStore?.ollamaModels.length}
		<a class="ml-auto" href="/settings">
			<Badge variant="warning">disconnected</Badge>
		</a>
	{/if}
</svelte:fragment>
```

Which looks like this:

![localhost_5173_sessions_y7ozps](https://github.com/user-attachments/assets/55ae77d7-d1e6-450c-8ea0-b196f7399ccf)

TBH, I don't love this solution but until we have a toast/flash alert system this would probably do.